hpc/LoadBalancer.cpp

```diff
@@ -35,7 +35,8 @@ void launch_hq_with_alloc_queue() {
 }
 
 const std::vector<std::string> get_model_names() {
-    HyperQueueJob hq_job("", false); // Don't start a client.
+    // Don't start a client, always use the default job submission script.
+    HyperQueueJob hq_job("", false, true); 
```

Comment:
How about encoding the meaning of `force_default_submission_script` as an empty string model name? Would save a single-purpose argument.
Technically nothing prevents users from defining a model named `""` though, so might actually better to go with your solution.

Reply:
Thanks for the feedback! This is the part that I'm also not super happy with...

I considered just using an empty model string at the beginning, but decided that it would be too error-prone:
- As you mentioned, a user might define a model named `""`.
- A user might have a file named `job_.sh` which would take precedence over the default `job.sh`.

The other alternative that I thought of would be to use `std::optional` and have `std::nullopt` indicate the default job script. However, I think that would needlessly complicate the logic, since you would have to properly deal with null values.

That's why I personally prefer the current version because it is simple, explicit and robust. I agree that having the extra argument is a bit awkward, but I think it's the lesser evil in this case.

Nevertheless, I don't mind changing it, so feel free to tell me what you think is best. 👍

Reply:
That makes sense, let's go with the lesser evil then! :)